src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -642,6 +691,53 @@
     </Touch>
   </Target>
 
+  <!--
+    Add Microsoft.Build.NoTargets and Microsoft.Build.Traversal SDK overrides, for all repos that utilize them,
+    except source-build-reference-packages repo, that builds these packages.
+
+    Also skip if building utility projects.
+
+    We need to override all SDK packages that are used by repos, to prevent NuGet SDK Resolver from trying to
+    restore these packages from online feeds. In offline build we only want to use local source-build feeds.
+    Therefore, we create a copy of repo's NuGet.config file, modify it, and pass it into repo build using
+    RestoreConfigFile property. MSBuild uses NuGetSdkResolver to restore missing SDKs, which doesn't honor
```

Comment:
> Which doesn't honor the RestoreConfifFile property

I wonder what the chances are that we can get this fixed by the team.

Reply:
Yes, I want this get fixed as well but it will likely require the env var to be set.

This could be combined with the msbuild sdk version override via an env var. Should be in the same code base.

If we reach out to the NuGet Team now, we might be able to get this implemented fairly quickly.

Reply:
There are existing issues with similar requests: https://github.com/NuGet/Home/issues/7855

However, we are now using a source-built SDK resolver to override SDKs and do not need the feature in NuGet or MSBuild. Additionally, we are using proper MSBuild property `RestoreConfigFile` that is fully supported.

The, rather verbose, comment I added for this target explains the reasoning behind the implementation. But I consider the implementation to be complete now. If we do hit bugs we will report them to other teams.

Reply:
See this issue which has more details: https://github.com/dotnet/source-build/issues/3781#issuecomment-1897375865

Middle to long term we would like to get rid off the custom msbuild sdk resolver. Its current design has several issues that are noted in the GH issue. We would like to just use the nuget sdk resolver and let it consume the live produced packages. That avoids a ton of custom infrastructure.